Regression/SLM/src/main/java/org/tribuo/regression/slm/SparseLinearModel.java

```diff
@@ -189,14 +201,149 @@ protected DimensionTuple scoreDimension(int dimensionIdx, SparseVector features)
 
     /**
      * Gets a copy of the model parameters.
+     *
      * @return A map from the dimension name to the model parameters.
      */
-    public Map<String,SparseVector> getWeights() {
+    public Map<String, SparseVector> getWeights() {
         SparseVector[] newWeights = copyWeights();
-        Map<String,SparseVector> output = new HashMap<>();
+        Map<String, SparseVector> output = new HashMap<>();
         for (int i = 0; i < dimensions.length; i++) {
-            output.put(dimensions[i],newWeights[i]);
+            output.put(dimensions[i], newWeights[i]);
         }
         return output;
     }
+
+    @Override
+    public OnnxMl.ModelProto exportONNXModel(String domain, long modelVersion) {
+        ONNXContext context = new ONNXContext();
+
+        // Build graph
+        OnnxMl.GraphProto graph = exportONNXGraph(context);
+
+        // Build model
+        OnnxMl.ModelProto.Builder builder = OnnxMl.ModelProto.newBuilder();
+        builder.setGraph(graph);
+        builder.setDomain(domain);
+        builder.setProducerName("Tribuo");
+        builder.setProducerVersion(Tribuo.VERSION);
+        builder.setModelVersion(modelVersion);
+        builder.setDocString(toString());
+        builder.addOpsetImport(ONNXOperators.getOpsetProto());
+        builder.setIrVersion(6);
+        return builder.build();
+    }
+
+    /**
+     * Builds a TensorProto containing the model weights.
+     * @param context The naming context.
+     * @return The weight TensorProto.
+     */
+    protected OnnxMl.TensorProto weightBuilder(ONNXContext context) {
```

Comment:
shouldn't these be factored out into `ONNXHelpers` or something - they're repeated almost verbatim from `AbstractLinearSGDModel`, and expect this won't be the last time we want to convert `DenseVector[]`/`DenseMatrix`/`DenseVector` into a `OnnxMl.TensorProto`.

Reply:
I could add another helper class into `tribuo-math` or add ONNX exporting functionality onto `Tensor` itself, but it can't live on `ONNXHelpers` as `DenseVector` et al aren't visible in `tribuo-core`. Unfortunately the SLM and SGD uses are all slightly different too, as SGD doesn't have a `DenseVector` for the biases it's packed onto the weight matrix.

Reply:
We're going to leave this until we have a third use-case and then see if we can extract common functionality from all three.